ndb/src/google/cloud/ndb/model.py

```diff
@@ -285,6 +285,50 @@ def _fix_up(self, cls, code_name):
         """
 
 
+class _BaseValue:
+    """A marker object wrapping a "base type" value.
+
+    This is used to be able to tell whether ``entity._values[name]`` is a
+    user value (i.e. of a type that the Python code understands) or a
+    base value (i.e of a type that serialization understands).
+    User values are unwrapped; base values are wrapped in a
+    :class:`_BaseValue` instance.
```

Comment:
Why would we be mixing the two kinds of values in the same mapping?  Seems like a recipe for unending pain to me.

Reply:
I agree with you, but there is a **lot** of complexity in this library. I'm trying to get a full picture before I pull this out.

Reply:
OK.  Is there a note to that effect in your migration docs, or should we create an issue for it?

Reply:
A note that I'm leaving in questionable choices from the original implementation? Or just specific to this type?

Reply:
Maybe we need a list of the oddballs to review later.

Reply:
OK. Sent out https://github.com/googleapis/google-cloud-python/pull/6289